AraEvent/AraEventCalibrator.cxx

```diff
@@ -1502,48 +1685,82 @@ Int_t AraEventCalibrator::numberOfPedestalValsInFile(char *fileName){
-        while(fAtriSampleADCVoltsConversion[dda][chan][block][sample][8]>1.0) sample = (sample - 2 + 32768)%32768;
+        if (stationId != 5){
+            while(fAtriSampleADCVoltsConversion[dda][chan][block][sample][8]>1.0) block = (block - neighboring_blk + blocks_per_dda)%blocks_per_dda; //based on Thomas's thesis p.69
+        } else {
+            if (sample%2==0 && chan>0) sample=(sample+1)%samples_per_dda;
+            while(fAtriSampleADCVoltsConversion[dda][chan][block][sample][8]>1.0) sample = (sample - neighboring_samp + samples_per_dda)%samples_per_dda;
+        }
 
         // Offset needs to be subtracted
-        double adcCounts = adcCountsIn;
+        double adcCounts = adcCountsIn + adc_offset;
+
         // Start ADC to voltage conversion
-        if(TMath::Abs(adcCounts)<500){
+        if(TMath::Abs(adcCounts)<high_adc_limit){
             // conversion factors for higher ADC values have strong errors, therefore we need the alternative calibration (see below)
             // RJN chnaged the below to remove calls to pow for code optimisation
             double modAdcCounts=adcCounts-fAtriSampleADCVoltsConversion[dda][chan][block][sample][7];
             //double modAdcCounts=adcCounts;//-fAtriSampleADCVoltsConversion[dda][chan][block][sample][7];
 
-            if(modAdcCounts>0){
+            Double_t fit_const; // Define the fit_const here (MK)
+            double adc_zero_def; // Define which value will be used to choose a positive or negative conversion
+            if (stationId != 5) {
+                fit_const = fAtriSampleADCVoltsConversion[dda][chan][block][sample][6];
+                adc_zero_def = adcCounts;
+            } else {
+                fit_const = 0.0;
+                adc_zero_def = modAdcCounts;
+            }
+
+            if(adc_zero_def>0){
                 //positive and negative values need different calibration constants
-                volts = 0.0
+                volts = fit_const
                     +modAdcCounts*fAtriSampleADCVoltsConversion[dda][chan][block][sample][0] 
                     +modAdcCounts*modAdcCounts*fAtriSampleADCVoltsConversion[dda][chan][block][sample][1] 
                     +modAdcCounts*modAdcCounts*modAdcCounts*fAtriSampleADCVoltsConversion[dda][chan][block][sample][2]; 
             }
             else{
-                volts = 0.0//fAtriSampleADCVoltsConversion[dda][chan][block][sample][6]         
+                volts = fit_const      
                     +modAdcCounts*fAtriSampleADCVoltsConversion[dda][chan][block][sample][3] 
                     +modAdcCounts*modAdcCounts*fAtriSampleADCVoltsConversion[dda][chan][block][sample][4]
                     +modAdcCounts*modAdcCounts*modAdcCounts*fAtriSampleADCVoltsConversion[dda][chan][block][sample][5];
             }
-            if(volts>800){
-        		volts=modAdcCounts;
-        	}
+
+            if (stationId == 5 && volts > 800) volts=modAdcCounts; //maybe need to put abs??? (MK)
```

Comment:
Why is there a specific check for A5?

Reply:
I don't follow this comment, sorry

Reply:
I guess, sometimes low ADC values (0 to +/- 400 ADC) is converted to unusually high voltage value like above 800mV which is too unusual compare to what we expect. So, If the conversion result is too high, I think this condition decides to use just ADC value instead of conversion results. And it is not happening in A2/3 in my observation.

Reply:
The more I think about this, the more I am uncomfortable with this. It suggests that for some reason the "high voltage" ADC conversion is not working. This is not something you should fix, but if you wouldn't mind, please open a new issue for it.

Reply:
Since it only affects A5, and this PR is not really targeted at affecting that calibration process, I would keep the changes that exclude A5.